Modules/rats.py

```diff
@@ -0,0 +1,183 @@
+    cache, allowing convenience method `Rats.get_rat` to return the instance
+    when called.
+    """
+
+    cache_by_id = {}
+    """Cache of rat objects by their UUID"""
+    cache_by_name = {}
+    """Cache of rat objects by rat name (str)"""
+
+    def __init__(self, uuid: UUID, name: str = None, platform: Platforms = Platforms.DEFAULT):
+        """
+        Creates a new rat
+
+        Args:
+            uuid (UUID):
+            name (str): rat's name
+            platform (Platforms): rat's platform
+        """
+        # set our properties
+        self._platform = platform
+        self._uuid = uuid
+        self._name = name
+        # and update the cache
+        if name and name not in Rats.cache_by_name:
+            # don't register duplicates
+            Rats.cache_by_name[name] = self
+        if uuid not in Rats.cache_by_id:
+            # don't register duplicates
+            Rats.cache_by_id[uuid] = self
+
+    @property
+    def uuid(self):
+        """
+        API id of the rat
+
+        Returns:
+            UUID
+        """
+        return self._uuid
+
+    @uuid.setter
+    def uuid(self, value) -> None:
+        """
+        Set the UUID of the rat.
+        accepts a uuid encoded string, or a uuid object.
+
+        Args:
+            value (UUID): new uuid for rat
+
+        Returns:
+            None
+        """
+        if isinstance(value, str):
+            LOG.debug(f"value was a string with data '{value}'")
+            uuid = UUID(value)
+            LOG.debug("parsed value into a valid UUID.")
+            self._uuid = uuid
+        elif isinstance(value, UUID):
+            self._uuid = value
+
+        else:
+            raise TypeError(f"expected type None or type str, got "
+                            f"{type(value)}")
+
+    @property
+    def name(self) -> str:
+        """
+        Rat's registered name
+
+        Returns:
+            str
+        """
+        return self._name
+
+    @name.setter
+    def name(self, value) -> None:
+        """
+        Sets the rat's registered name
+
+        Args:
+            value (str): name
+
+        Returns:
+            None
+        """
+        if isinstance(value, str):
+            self._name = value
+        else:
+            raise TypeError(f"expected str, got {type(value)}")
+
+    @property
+    def platform(self) -> Platforms:
+        """
+        The Rats platform
+
+        Returns:
+            Platforms: The platform the rat is registered on
+        """
+        return self._platform
+
+    @platform.setter
+    def platform(self, value: Platforms) -> None:
+        """
+        Sets the platform for a given rat
+
+        Args:
+            value (Platforms): new platform
+        """
+        if isinstance(value, Platforms):
+            self._platform = value
+        else:
+            raise TypeError(f"Expected a {Platforms} object, got type {type(value)}")
```

Comment:
Similar here as above: assuming that `Platforms` is a enum, you'll get `<enum 'Platforms'>` here. Platforms.__name__ is what you want.

Reply:
Platforms is an enum, though again it should be enough information to point us at the fault for thorough debugging and bugfixing. Calling a magic method for something like this seems needless